src/probnum/statespace/discrete_transition.py

```diff
@@ -334,14 +351,22 @@ def _forward_rv_classic(
         new_cov = H @ crosscov + _diffusion * R
         info = {"crosscov": crosscov}
         if compute_gain:
-            gain = scipy.linalg.solve(new_cov.T, crosscov.T, assume_a="sym").T
+            if config.lazy_linalg:
+                gain = (new_cov.T.inv() @ crosscov.T).T
+            else:
+                gain = scipy.linalg.solve(new_cov.T, crosscov.T, assume_a="sym").T
```

Comment:
I am not sure about the overhead the creation of the `linops.Matrix` object and the subsequent "densification" would create. I think this is best handled in a separate PR in which we work on creating a unified interface for the `LinearOperator`s. (And this should - as discussed - happen rather sooner than later).

Reply:
I am not sure whether there will be too much overhead, and I don't know what you mean by densification (the input would be dense already? I am talking about having the dense matrix as input to linops.Matrix and then this would be it I guess).

But I think it is fair to do attempt this in a different PR. Just wanted to bring up that I think this is a direction worth going :)